Sources/HackernewsNLU/Controller.swift

```diff
@@ -189,78 +176,84 @@ public class Controller {
     ///
     /// - parameter completion: Completion closure invoked on success
     /// - parameter failure:    Failure closure invoked on error
-    func callNLUApi(newsArticle article: NewsArticles,completion: @escaping (JSON)->(), failure: @escaping (String) -> Void) -> Void {
-        let path = "/natural-language-understanding/api/v1/analyze?version=2017-02-27"
-        guard let username = nluCreds["username"] else {
-            failure("No username for NLU service")
-            return
+    func callNLUApi(newsArticle article: NewsArticles, completion: @escaping (SwiftyJSON.JSON) -> Void, failure: @escaping (String) -> Void) {
+        var service: NaturalLanguageUnderstanding? = nil
+        func checkValidity(username: String?, password: String?) -> Bool {
+            return (!(username == nil) && !(username == (username == nil ? nil : ""))) && (!(password == nil) && !(password == (password == nil ? nil : "")))
         }
-        guard let password = nluCreds["password"] else {
-            failure("No password for NLU service")
-            return
+        func checkValidity(apikey: String?) -> Bool {
+            return !(apikey == nil) && !(apikey == (apikey == nil ? nil : ""))
         }
-        var requestOptions: [ClientRequest.Options] = []
-        var headers = [String:String]()
-        let theBody: JSON = [
-            "url": article.getLink(),
-            "features":[
-                "concepts":   [ "limit":10],
-                "categories":  [ "limit":10],
-                "emotion":  [ "limit":10],
-                "entities":  [ "limit":10],
-                "sentiment":  [ "limit":10],
-                "keywords":  [ "limit":10]
-            ]
-        ]
-        var theData = Data()
-        do {
-            try theData = theBody.rawData(options: JSONSerialization.WritingOptions())
-        }
-        catch {
-            print("error")
+        let apikeyValid = checkValidity(apikey: nluCreds["apikey"])
+        let userpassValid = checkValidity(username: nluCreds["username"], password: nluCreds["password"])
+        if apikeyValid && userpassValid {
+            failure("You provided both a username and password, as! well as! an API key. Please provide only one.")
```

Comment:
why the exclamation points? `as! well as! an`

Reply:
will fix